integration/src/shopify-token-exchange/my-after-auth-handler.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  AccessMode,
+  ShopifyAuthTokenExchangeAfterHandler,
+  ShopifyAuthTokenExchangeAfterHandlerParams,
+  ShopifyTokenExchangeService,
+} from '@nestjs-shopify/auth';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class MyAfterAuthHandler
+  implements ShopifyAuthTokenExchangeAfterHandler
+{
+  constructor(
+    private readonly tokenExchangeService: ShopifyTokenExchangeService,
+  ) {}
+  async afterAuth({
+    session,
+    sessionToken,
+  }: ShopifyAuthTokenExchangeAfterHandlerParams) {
+    if (session.isOnline) {
+      try {
+        const offlineSession = await this.tokenExchangeService.exchangeToken(
+          sessionToken,
+          session.shop,
+          AccessMode.Offline,
+        );
+        console.log('Offline session:', offlineSession);
+      } catch (e) {
+        /* empty */
+      }
```

Comment:
We should be able to avoid exposing this to the users. The `AfterAuthHandler` should only be used to "setup" things after authorization. Not to "finish" authorization.

Based on the `options` provided, the **provider that calls** `MyAfterAuthHandler` should perform the `exchangeToken` call.